K/KaHyPar/build_tarballs.jl

```diff
@@ -0,0 +1,43 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "KaHyPar"
+version = v"1.0.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/arbenson/kahypar.git", "d64498068ab745a09a94391e746d35853fc3eef3"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd kahypar/
+git submodule update --init --recursive
+mkdir build
+cd build
+cmake .. -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_BUILD_TYPE=Release
+make
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Linux(:x86_64, libc=:musl),
+    Linux(:x86_64, libc=:glibc)
+]
+
```

Comment:
Note that the audit issued the following warning in the [compilation log](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=2058&view=logs&jobId=9ec5b5b9-8d5c-5863-d149-dcf2a420ecd9&j=9ec5b5b9-8d5c-5863-d149-dcf2a420ecd9&t=9c05869b-7e06-50f7-30c4-0fae9af3d278):
```
┌ Warning: /agent/_work/1/s/K/KaHyPar/build/x86_64-linux-musl/dMPwiDpS/destdir/lib64/libkahypar.so contains std::string values!  This causes incompatibilities across the GCC 4/5 version boundary.  To remedy this, you must build a tarball for both GCC 4 and GCC 5.  To do this, immediately after your `platforms` definition in your `build_tarballs.jl` file, add the line:
│ 
│     platforms = expand_cxxstring_abis(platforms)
└ @ BinaryBuilder /depot/packages/BinaryBuilder/4fMlX/src/auditor/compiler_abi.jl:189
```
```suggestion
platforms = expand_cxxstring_abis(platforms)
```
Also, any particular reason why you built only for these two platforms?

Reply:
i686 was taking > 24 hours without finishing (compared to 20 minutes on x86_64), and I would like to have something. I wanted to get something working to start building the package that would use this _JLL.

Reply:
> i686 was taking > 24 hours without finishing (compared to 20 minutes on x86_64)

That's very strange.  Here we use a couple of tricks to speedup compilation and recompilation: parallel building and ccache, they're often very effective.

Anyway, expanding cxxstring abis as explained above is important